logic/atrium-logic/src/commonMain/kotlin/ch/tutteli/atrium/logic/impl/DefaultAnyAssertions.kt

```diff
@@ -18,8 +17,17 @@ class DefaultAnyAssertions : AnyAssertions {
     override fun <T> toBe(container: AssertionContainer<T>, expected: T): Assertion =
         container.createDescriptiveAssertion(TO_EQUAL, expected) { it == expected }
 
-    override fun <T> notToBe(container: AssertionContainer<T>, expected: T): Assertion =
-        container.createDescriptiveAssertion(NOT_TO_EQUAL, expected) { it != expected }
+    override fun <T> notToBe(container: AssertionContainer<T?>, expected: T?): Assertion {
+
+        if (container.maybeSubject.getOrElse { null } == null || expected == null) {
```

Comment:
We are only interested in the case where `expected` is ` null`  not where the subject is null. In the end we want more or less the same report for 
```
expect(null as Int?).notToEqualNull { ... }
``` 
and
``` 
expect(null as Int?).notToEqual(null)
```
(but not for `expect(null as Int?).notToEqual(2)` 
However, I just realised, that we would need to break notToBe and expect `subType: KClass<T>`  in addition in order that you can report the type. My bad, we won't do this at this point. so please revert all changed related to this requirement (all changes in this file basically)

Reply:
Okay no problem so we are just keeping the translation and initial `notToEqualNull` ?

Reply:
I think the code as it was after commit d02abb19c6a69e883fea17fd71536dac25096cd7 was everything we needed. I can also cherry-pick from there I totally get that it starts to get confusing now

Reply:
Yeah that would be helpful thank you